pkg/actuators/machine/loadbalancers.go

```diff
@@ -38,41 +38,19 @@ func registerWithClassicLoadBalancers(client awsclient.Client, names []string, i
 
 func registerWithNetworkLoadBalancers(client awsclient.Client, names []string, instance *ec2.Instance) error {
 	klog.V(4).Infof("Updating network load balancer registration for %q", *instance.InstanceId)
-	lbNames := make([]*string, len(names))
-	for i, name := range names {
-		lbNames[i] = aws.String(name)
-	}
-	lbsRequest := &elbv2.DescribeLoadBalancersInput{
-		Names: lbNames,
-	}
-	lbsResponse, err := client.ELBv2DescribeLoadBalancers(lbsRequest)
+	targetGroups, err := gatherLoadBalancerTargetGroups(client, names)
 	if err != nil {
-		klog.Errorf("Failed to describe load balancers %v: %v", names, err)
 		return err
 	}
-	// Use a map for target groups to get unique target group entries across load balancers
-	targetGroups := map[string]*elbv2.TargetGroup{}
-	for _, loadBalancer := range lbsResponse.LoadBalancers {
-		klog.V(4).Infof("Retrieving target groups for load balancer %q", *loadBalancer.LoadBalancerName)
-		targetGroupsInput := &elbv2.DescribeTargetGroupsInput{
-			LoadBalancerArn: loadBalancer.LoadBalancerArn,
-		}
-		targetGroupsOutput, err := client.ELBv2DescribeTargetGroups(targetGroupsInput)
-		if err != nil {
-			klog.Errorf("Failed to retrieve load balancer target groups for %q: %v", *loadBalancer.LoadBalancerName, err)
-			return err
-		}
-		for _, targetGroup := range targetGroupsOutput.TargetGroups {
-			targetGroups[*targetGroup.TargetGroupArn] = targetGroup
-		}
-	}
+
 	if klog.V(4).Enabled() {
-		targetGroupArns := make([]string, 0, len(targetGroups))
-		for arn := range targetGroups {
-			targetGroupArns = append(targetGroupArns, fmt.Sprintf("%q", arn))
+		var targetGroupArns []string
+		for _, targetGroup := range targetGroups {
+			targetGroupArns = append(targetGroupArns, fmt.Sprintf("%q", *targetGroup.TargetGroupArn))
```

Comment:
Can you make sure there's a test to make sure we aren't capturing the pointer here. IE if we have multiple ARNs then this should return multiple arns and not just the same one over and other

Reply:
I refactored this part of the code, as it served no particular purpose, and was a plain c/p from registerLB. Testing higher log level enclosures requires changes to make file to include selector for methods which needs such testing, and is generally useless as here we test plain formatting for log messages done wrong.

Reply:
Perhaps make a method of this that can be tested to ensure it joins the strings correctly?

So then the log would look lie
```
klog.Infof("Unregistering instance %q registered by ip from target groups: %v", *instance.InstanceId, getTargetGroupARNs(targetGroupArns, ","))
```

I'm just concerned you might fall foul of something like https://eli.thegreenplace.net/2019/go-internals-capturing-loop-variables-in-closures/

Reply:
Sorry. This part was not refactored. I'm not going to make a method. 2 questions: does it add value? No. Does it introduce coplexity? Yes. This code is a legacy we'd like to avoid using as far as I can tell. Let me quickly push a fix.

Reply:
> This part was not refactored.

Yes it was, the logic here has changed from looking up a string in a map, to getting a field from a pointer to a struct. While the previous implementation is not susceptible to loop capturing issues, when you take a value from a pointer, it can be. I think we are probably ok here looking at it, but I don't think adding a small helper function adds complexity, if anything, it would make this code easier to read

Reply:
All fields in AWS API structures are pointers. We need nil check on each and every of them, wherever we iterate over lists from response, or access a value. ARN is a unique identifier. IMO, if this log message fails, we open issue in AWS api.